src/System.Management.Automation/engine/TypeTable.cs

```diff
@@ -3448,12 +3453,12 @@ internal static void ProcessMemberSetData(ConcurrentBag<string> errors, string t
         private static void ProcessStandardMembers(
             ConcurrentBag<string> errors,
             string typeName,
-            IEnumerable<TypeMemberData> standardMembers,
-            IEnumerable<PropertySetData> propertySets,
+            Dictionary<string, TypeMemberData>.ValueCollection standardMembers,
+            List<PropertySetData> propertySets,
```

Comment:
`IReadOnlyCollection<T>` is probably sufficient here, without over-specifying the type

Reply:
@rjmholt Could you please clarify that do you mean?

Reply:
But `IReadOnlyCollection<T>` use a non-ValueType enumerator which may cause a heap allocation. Anyway, this is private code.

Reply:
The possibility of the single allocation is worth the factorability to my mind. Passing `Dictionary<K, V>.ValueCollection` is really not ideal.

Private code or not, the method signature is a contract, and the method doesn't have a strong requirement on its inputs.

Reply:
I've refactored to `Dictionary<string, TypeMemberData>` which matches the type of class property `StandardMembers`.